### PR TITLE
Adds the twitterize proc, so that the bird may live forever

### PR DESCRIPTION
**Pull request by Incoming5643**
Twitterize takes a list of strings and turns it into a JSON file ([example](http://pastebin.com/raw/5F1Hr2P7)) of 140 characters or less strings.

If you wondering why this exists a little ways back I launched a pet project twitter bot that works through poly's speech buffer. (https://tgstation13.org/phpBB/viewtopic.php?f=2&t=8823)

However I didn't do anything to automate how twitter!poly got the phrases from ss13!poly. Reading the .sav files directly is a little beyond me right now, and doing it like this I can ""enforce"" a standard for input if other codebases/downstreams want to give me a way to keep track of their birds so I can add them to the growing gestalt entity that threatens to one day devour us all. Between this and what I'm doing on my end, twitter!poly should be able to exist until the heat death of the universe.

Going forward this could also be used for other humorous collections of spessmen creativity, like a twitter of station announcements or newscaster stories.

### CODE REVIEW

File: code/__HELPERS/text.dm [hunk 1]
```diff
@@ -438,3 +438,85 @@ var/list/binary = list("0","1")
...
+//Takes a list of values, sanitizes it down for readability and character count,
+//then exports it as a plaintext file at data/npc_saves/[filename].txt.
+//As far as SS13 is concerned this is write only data. You can't change something
+//in the txt file and have it be reflected in the in game item/mob it came from.
+//(That's what things like savefiles are for) Note that this list is not shuffled.
+/proc/twitterize(list/proposed, filename, cullshort = 0, storemax = 1000)
+	if(!islist(proposed) || !filename || storemax > proposed.len)
+		return
+
+	var/list/nope = list("www.", "http", "@") //Note the ommission of #, hashtags are fine
```
Comment (Iamgoofball) [inline code/__HELPERS/text.dm]: we should probably have a file with blacklisted words too for server owners if we want to blacklist slurs and shit easily

Comment (Incoming5643) [inline code/__HELPERS/text.dm]: Well all that would actually do is blacklist the output to people using it to run twitter bots. I already have a little slur blacklist running on the other end of this here operation, but I guess people could want to do it on this side too for like... Public Relations?

**Comment by ShadowDeath6** [comment]
>poly should be able to exist until the heat death of the universe

finally, a pr for me!
:+1:

File: code/__HELPERS/text.dm (hunk 1 above)
Comment (lordpidey) [inline code/__HELPERS/text.dm]: add TLDs to this list.

File: code/__HELPERS/text.dm [hunk 2]
```diff
@@ -438,3 +438,95 @@ var/list/binary = list("0","1")
...
+//Takes a list of values, sanitizes it down for readability and character count,
+//then exports it as a plaintext file at data/npc_saves/[filename].txt.
+//As far as SS13 is concerned this is write only data. You can't change something
+//in the txt file and have it be reflected in the in game item/mob it came from.
+//(That's what things like savefiles are for) Note that this list is not shuffled.
+/proc/twitterize(list/proposed, filename, cullshort = 0, storemax = 1000)
+	if(!islist(proposed) || !filename || storemax < proposed.len)
+		return
+
+	var/list/nope = list("www.", "http", "@", ".com", ".org", ".net", ".edu") //Note the ommission of #, hashtags are fine
```
Comment (RemieRichards) [inline code/__HELPERS/text.dm]: use a regex to match weblink-like things, very common pattern.

File: code/__HELPERS/text.dm [hunk 3]
```diff
@@ -445,35 +445,39 @@ var/list/binary = list("0","1")
 //in the txt file and have it be reflected in the in game item/mob it came from.
 //(That's what things like savefiles are for) Note that this list is not shuffled.
 /proc/twitterize(list/proposed, filename, cullshort = 0, storemax = 1000)
-	if(!islist(proposed) || !filename || storemax > proposed.len)
+	if(!islist(proposed) || !filename || storemax < proposed.len)
 		return
 
-	var/list/nope = list("www.", "http", "@") //Note the ommission of #, hashtags are fine
+	var/list/nope = list("www.", "http", "@", ".com", ".org", ".net", ".edu") //Note the ommission of #, hashtags are fine
```
Comment (Incoming5643) [inline code/__HELPERS/text.dm]: yes it's a losing fight, which is why I didn't even bother initially

**Comment by optimumtact** [comment]
PLEASE PR WORKING CODE

File: code/__HELPERS/text.dm [hunk 4]
```diff
@@ -438,3 +438,91 @@ var/list/binary = list("0","1")
...
+	if(!isemptylist(oldentries))
+		for(var/string in accepted) //The fifth and final gate
+			for(var/old in oldentries)
+				if(string == old)
+					oldentries.Remove(old) //Line's position in line is "refreshed" until it falls off the in game radar
+					break
+
+	var/list/finalized = list()
+	for(var/list/L in list(accepted, oldentries)) //we keep old and unreferenced phrases near the bottom for culling
+		finalized += listclearnulls(L.Copy())
```
Comment (Incoming5643) [inline code/__HELPERS/text.dm]: Though accepted is definitely populated and oldentries sometimes is I've yet to find a way to transfer them to finalized and pass them through the mutator listclearnulls() that doesn't blank out the whole of finalized.

Besides this everything works.

**Comment by Incoming5643** [comment]
After a lot of heartache I've gotten this to a fully functional state.

**Comment by karlnp** [comment]
merge pls so i can iterate

**Comment by Incoming5643** [comment]
Just a note to anyone who might want to port this: While this works the filtering isn't perfect yet, a follow up PR is planned but I'm just letting it run for a while to collect info on what's not parsing 100% correctly.

**Pull request merged**